src/gas/GasFeeController.ts

```diff
@@ -247,18 +298,17 @@ export class GasFeeController extends BaseController<typeof name, GasFeeState> {
           suggestedMaxPriorityFeePerGas,
           suggestedMaxFeePerGas,
         );
-      } catch (error) {
-        try {
-          estimates = await this.fetchLegacyGasPriceEstimate(this.ethQuery);
-        } catch (error2) {
-          throw new Error(
-            `Gas fee/price estimation failed. Message: ${error2.message}`,
-          );
-        }
+        gasEstimateType = GAS_ESTIMATE_TYPES.FEE_MARKET;
+      } else if (isMainnet) {
+        estimates = await this.fetchLegacyGasPriceEstimates();
+        gasEstimateType = GAS_ESTIMATE_TYPES.LEGACY;
+      } else {
+        throw new Error('Main gas fee/price estimation failed. Use fallback');
       }
-    } else {
+    } catch (error) {
       try {
-        estimates = await this.fetchLegacyGasPriceEstimate(this.ethQuery);
+        estimates = await this.fetchEthGasPriceEstimate(this.ethQuery);
+        gasEstimateType = GAS_ESTIMATE_TYPES.ETH_GASPRICE;
       } catch (error2) {
```

Comment:
if we are not tracking `error` from `fetchGasEstimates` or `fetchLegacyGasPriceEstimates` we can just `} catch {` and rename `error2` to `error`

Reply:
done